client/src/cli/__init__.py

```diff
@@ -1,27 +1,68 @@
 import logging
-import os
 import sys
 
-from .parser import main_parser
+import pydantic
+from pydantic_settings import CliSettingsSource
 
+import cli.parser
+from cli.types import Settings
 
+
+def _configure_logging(settings: Settings) -> None:
+    logging.getLogger().setLevel(
```

Comment:
setLevel can deal with strings, see https://docs.python.org/3/library/logging.html#logging.Logger.setLevel.

We can supply the default from the argparse default, which is more sensible imo.